packages/cerebral/src/Controller.js

```diff
@@ -166,7 +166,30 @@ class Controller extends FunctionTree {
       }, {})
     }
 
-    this.runTree(name, signal, payload)
+    this.runTree(name, signal, payload, (error) => {
+      if (error) {
+        const signalPath = error.execution.name.split('.')
+        const signalCatch = signalPath.reduce((currentModule, key, index) => {
+          if (index === signalPath.length - 1) {
+            return currentModule.signals[key].catch
+          }
+
+          return currentModule ? currentModule.modules[key] : undefined
+        }, this.module)
+
+        if (!signalCatch) {
+          throw error
+        }
+
+        if (Array.isArray(signalCatch)) {
+          this.runSignal('catch', signalCatch, error.payload)
+        } else if (error.name in signalCatch) {
```

Comment:
suggest allowing primitives for consistency, not just sequences as arrays (if not that way already). commented here since this is the snippet that raised my suspicion.